### PR TITLE
fix(chat): removes sidebar delete and import buttons if only shared items exist (Issue #750)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

 Removes sidebar delete and import buttons if only shared items exist.

Issues:

- Issue: https://github.com/epam/ai-dial-chat/issues/750

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

File: apps/chat/src/store/conversations/conversations.selectors.ts [hunk 1]
```diff
@@ -255,6 +255,20 @@ export const selectAreSelectedConversationsExternal = createSelector(
...
+    return {
+      externalConversations: conversations.filter((prompt) =>
+        isEntityOrParentsExternal(state, prompt, FeatureType.Chat),
+      ),
+      externalFolders: folders.filter((folder) =>
+        isEntityOrParentsExternal(state, folder, FeatureType.Chat),
+      ),
+    };
+  },
+);
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.selectors.ts]: Why do we do all this check on UI?
Let's memorise it in selector `selectDoesAnyMyItemExists` which returns boolean instead
```suggestion
export const selectDoesAnyMyItemExist = createSelector(
  [selectFolders, selectConversations],
  (folders, conversations) => {
    const conversationRootId = getConversationRootId();
    return (
      conversations.some((conv) => conv.id.startsWith(conversationRootId)) ||
      folders.some((folder) => folder.id.startsWith(conversationRootId))
    );
  },
);
```

File: apps/chat/src/store/prompts/prompts.selectors.ts [hunk 2]
```diff
@@ -211,6 +215,20 @@ export const hasExternalParent = createSelector(
...
+      externalPrompts: prompts.filter((prompt) =>
+        isEntityOrParentsExternal(state, prompt, FeatureType.Prompt),
+      ),
+      externalFolders: folders.filter((folder) =>
+        isEntityOrParentsExternal(state, folder, FeatureType.Prompt),
+      ),
+    };
+  },
+);
+
```
Comment (IlyaBondar) [inline apps/chat/src/store/prompts/prompts.selectors.ts]: Why do we do all this check on UI?
Let's memorise it in selector `selectDoesAnyMyItemExists` which returns boolean instead
```suggestion
export const selectDoesAnyMyItemExist = createSelector(
  [selectFolders, selectPrompts],
  (folders, prompts) => {
    const promptRootId = getPromptRootId();
    return (
      prompts.some((prompt) => prompt.id.startsWith(promptRootId )) ||
      folders.some((folder) => folder.id.startsWith(promptRootId ))
    );
  },
);
```

**Review by IlyaBondar — CHANGES_REQUESTED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-886.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**